### PR TITLE
Fixing warnings issues with Apple Compiler/Xcode (old)

### PR DESCRIPTION
**Pull request by jbr-smtg**
Hello all,

There are a couples of occurrence of a warnings with the Apple Compiler. The IDE Xcode suggests the fix in this pull request.

Thanks in advance!

### CODE REVIEW

**Comment by grasci-arm** [comment]
Can one of the admins verify this patch?

**Comment by jbr-smtg** [comment]
I'm not sure how to fix the issue "Error: CMSIS/Core/Include/cmsis_gcc.h:4:Please increment file version."...

**Comment by JonatanAntoni** [comment]
Hi @jbr-smtg,

this looks like a false positive in the checker. I can see you incremented the file version as expected. Not sure why the checker doesn't recognize that.

Cheers,
Jonatan

**Comment by christophe0606** [comment]
Since this pull request is done in the context of CMSIS-DSP, I'll check the changes are not causing issues to build with our gcc on Cortex-M.

And if this check is ok, I'll let @JonatanAntoni manage the next steps

File: CMSIS/Core/Include/cmsis_gcc.h [hunk 1]
```diff
@@ -296,7 +296,7 @@ __STATIC_FORCEINLINE uint32_t __REV(uint32_t value)
 #else
   uint32_t result;
 
-  __ASM ("rev %0, %1" : __CMSIS_GCC_OUT_REG (result) : __CMSIS_GCC_USE_REG (value) );
+	__ASM ("rev %w0, %w1" : __CMSIS_GCC_OUT_REG (result) : __CMSIS_GCC_USE_REG (value) );
```
Comment (JonatanAntoni) [inline CMSIS/Core/Include/cmsis_gcc.h:299]: Don't use tabs. Replace tabs with two spaces, please.

**Review by JonatanAntoni — CHANGES_REQUESTED** [review]

**Comment by jbr-smtg** [comment]
Somehow I might have done something wrong; it seems that there is now even more red than before... 😵‍💫

**Comment by JonatanAntoni** [comment]
Yes, you need to rebase your changes on top of develop branch. And during rebase you need to squash all changes into a single commit. After that you won't be able to push fast-forward to your branch anymore; you need to force-push (--force-with-lease). 

Once your PR only consists of a single commit based on latest develop (containing my fix) the header check should pass.

I've seen issues with the CodeQL check as well. These are caused by build issues. Not sure if these issues are introduced by your changes; we'd need to check that.

**Comment by jbr-smtg** [comment]
I have closed this pull request and created a new one:
https://github.com/ARM-software/CMSIS_5/pull/1325

This seemed a lot easier to fix the issues...

**Pull request closed without merging**